src/routes/search/components/SearchModal.js

```diff
@@ -51,27 +52,45 @@ class SearchModal extends React.Component<PropsType, StateType> {
 
     // find all categories whose titles include the filter text and sort them lexicographically
     const categoriesWithTitle = categoriesArray
-      .filter(category => normalizeSearchString(category.title).includes(normalizedFilter))
-      .sort((category1, category2) => category1.title.localeCompare(category2.title))
+      .filter(category => normalizeSearchString(category.title).includes(normalizedFilter) && !category.isRoot())
+      .map((category: CategoryModel): ListEntryType => ({
+        model: {
+          title: category.title,
+          thumbnail: category.thumbnail,
+          path: category.path
+        },
+        subCategories: []
+      })
+
+      )
+      .sort((category1, category2) => category1.model.title.localeCompare(category2.model.title))
 
     // find all categories whose contents but not titles include the filter text and sort them lexicographically
+    let contentWithoutHtml = []
+    const parser = new Parser({ ontext (data: string) { contentWithoutHtml.push(data) } })
     const categoriesWithContent = categoriesArray
-      .filter(category => !normalizeSearchString(category.title).includes(normalizedFilter))
-      .filter(category => normalizeSearchString(category.content).includes(normalizedFilter))
-      .sort((category1, category2) => category1.title.localeCompare(category2.title))
+      .filter(category => !normalizeSearchString(category.title).includes(normalizedFilter) && !category.isRoot())
+      .map((category: CategoryModel): ListEntryType => {
+        contentWithoutHtml = []
+        parser.write(category.content)
+        parser.end()
```

Comment:
If you call parser.end() within the map function then I would assume that the parser also gets initialized here.

Reply:
I am kind of worried that the Parser initialization will impact the performance at least visibly. But I will test it shortly.

Reply:
Seems to not matter